db-client-java/src/main/java/com/eventstore/dbclient/EventStoreDBClient.java

```diff
@@ -142,84 +197,166 @@ public CompletableFuture<StreamMetadata> getStreamMetadata(String streamName, Re
+     * one until it reaches the end of the stream. From there, the server will notify the handler whenever a new event
+     * appears.
+     * @param streamName stream's name.
+     * @param listener consumes a subscription's events.
+     * @param options a subscription request's options.
+     * @return a subscription handle.
+     */
     public CompletableFuture<Subscription> subscribeToStream(String streamName, SubscriptionListener listener, SubscribeToStreamOptions options) {
         if (options == null)
             options = SubscribeToStreamOptions.get();
 
-        if (!options.hasUserCredentials())
-            options.authenticated(this.credentials);
-
-        return new SubscribeToStream(this.client, streamName, listener, options).execute();
+        return new SubscribeToStream(this.getGrpcClient(), streamName, listener, options).execute();
     }
 
+    /**
+     * Subscriptions allow you to subscribe to $all stream and receive notifications about new events added to the stream.
+     * You provide an even handler and an optional starting point to the subscription. The handler is called for each
+     * event from the starting point onward. If events already exist, the handler will be called for each event one by
+     * one until it reaches the end of the stream. From there, the server will notify the handler whenever a new event
+     * appears.
+     * @param listener consumes a subscription's events.
+     * @return a subscription handle.
+     */
     public CompletableFuture<Subscription> subscribeToAll(SubscriptionListener listener) {
         return this.subscribeToAll(listener, SubscribeToAllOptions.get());
     }
 
+    /**
+     * Subscriptions allow you to subscribe to $all stream and receive notifications about new events added to the stream.
+     * You provide an even handler and an optional starting point to the subscription. The handler is called for each
+     * event from the starting point onward. If events already exist, the handler will be called for each event one by
+     * one until it reaches the end of the stream. From there, the server will notify the handler whenever a new event
+     * appears.
+     * @param listener consumes a subscription's events.
+     * @param options subscription to $all request's options.
+     * @return a subscription handle.
+     */
     public CompletableFuture<Subscription> subscribeToAll(SubscriptionListener listener, SubscribeToAllOptions options) {
         if (options == null)
             options = SubscribeToAllOptions.get();
 
-        if (!options.hasUserCredentials())
-            options.authenticated(this.credentials);
-
-        return new SubscribeToAll(this.client, listener, options).execute();
-    }
-
+        return new SubscribeToAll(this.getGrpcClient(), listener, options).execute();
+    }
+
+    /**
+     * Soft deletes a given stream.
```

Comment:
We want to move away from soft/hard delete to avoid confusion, instead we want "delete" and "tombstone"

```suggestion
     * Deletes a given stream.
```